### PR TITLE
docker-compose pull SERVICE should not pull SERVICE's dependencies

### PR DESCRIPTION
**Pull request by cweagans**
Fixes https://github.com/docker/compose/issues/1752

### CODE REVIEW

File: compose/project.py [hunk 1]
```diff
@@ -335,7 +335,10 @@ def _get_convergence_plans(self, services, strategy):
         return plans
 
     def pull(self, service_names=None, ignore_pull_failures=False):
-        for service in self.get_services(service_names, include_deps=True):
+        pull_dependencies = True
+        if service_names:
+            pull_dependencies = False
+        for service in self.get_services(service_names, include_deps=pull_dependencies):
```
Comment (dnephin) [inline compose/project.py]: Let's just do `include_deps=False` here. If there are no services_names it's already going to pull everything, so the behaviour should be the same.

**Comment by dnephin** [comment]
Thanks for fixing this!

File: compose/project.py (hunk 1 above)
Comment (cweagans) [inline compose/project.py]: :+1:

**Comment by cweagans** [comment]
@dnephin Pushed a new commit per your comment earlier.

**Comment by dnephin** [comment]
LGTM (assuming jenky is green)

**Comment by cweagans** [comment]
Is there a process for backporting fixes to an earlier version of Compose? Or is it more of an "upgrade to get new stuff and bug fixes" kind of thing?

**Comment by mnowster** [comment]
@cweagans As a general guide it's more of an upgrade to get fixes. Imbetween major releases we'll do minor ones which include high priority fixes rather than making people wait for 2months for the next major release.

This will go into the 1.5 release. 

LGTM.

**Comment by cweagans** [comment]
@mnowster Roger that. Thanks for the merge.

**Pull request merged**